var/spack/repos/builtin/packages/py-meson-python/package.py

```diff
@@ -34,7 +31,7 @@ class PyMesonPython(PythonPackage):
     depends_on("py-pyproject-metadata@0.5:", type=("build", "run"))
     depends_on("py-tomli@1:", when="@0.11: ^python@:3.10", type=("build", "run"))
     depends_on("py-tomli@1:", when="@:0.10", type=("build", "run"))
-    depends_on("py-typing-extensions@3.7.4:", when="@0.12: ^python@:3.9", type=("build", "run"))
+    depends_on("py-typing-extensions@3.7.4:", when="@0.12 ^python@:3.9", type=("build", "run"))
     depends_on("py-typing-extensions@3.7.4:", when="@:0.11 ^python@:3.7", type=("build", "run"))
```

Comment:
Needs a dep on setuptools 60+ when python 3.12+

Reply:
I should have commented on that: I decided not to request that because I expect that to disappear before the Python 3.12.0 release (Meson has a zero-dependencies policy, and IIRC meson-python added this as a workaround for something in Meson). And I assume that Spack would not deal with Python 3.12 before the official release date.

That said, it can't hurt to add it with a comment that it can likely be removed once Python 3.12 is out.

Reply:
Up to y'all. Someone may want to install old meson-python with new python in the future.

Reply:
> Someone may want to install old meson-python with new python in the future.

There's nothing in meson-python itself that this needs this, it's in Meson. But that perhaps makes it more likely - and I don't think there's a need to forbid that combo, because users may have an environment with older Meson that's not used for building a Python package. So I'm fine with adding the dependency.

Reply:
I'm neutral on this, and primarily interested in being a useful source of contextual information. :)

> Up to y'all. Someone may want to install old meson-python with new python in the future.

The version of meson-python isn't strictly relevant -- what matters here is:
- the version of python used for meson-python
- the version of meson, without regard for the version of python used

meson-python will pass its own python executable to meson, and tell meson to build for that python executable. And meson needs that python executable to have distutils, or be updated to stop using that on new pythons.

meson can depend on a different environment's copy of python, of any version.

So, meson-python depends on:
- python
- py-tomli (when=...)
- py-pyproject-metadata (when=...)
- py-typing-extensions (when=...)
- osdep: ninja
- osdep: meson
- py-setuptools (if I understand how specs work correctly): `when="^python@3.12: ^meson@:future-version",`


But in that last, very interesting case, neither the relevant version of python, nor the relevant version of meson, has been released. One could ask "in that case, who cares". On the other hand, one could say "it is technically correct and maximally descriptive, to care and include correct annotations".